clients/gitlabrepo/repo_test.go

```diff
@@ -157,9 +157,13 @@ func TestRepoURL_DetectGitlab(t *testing.T) {
 		if tt.flagRequired && os.Getenv("TEST_GITLAB_EXTERNAL") == "" {
 			continue
 		}
-		g := DetectGitLab(tt.repouri)
-		if g != tt.expected {
-			t.Errorf("got %s isgitlab: %t expected %t", tt.repouri, g, tt.expected)
+		g, err := MakeGitlabRepo(tt.repouri)
+		if (g != nil) != (err == nil) {
```

Comment:
should this just be `if err != nil` ? or are we expecting some calls to error?

Reply:
There should be an error whenever the result is nil, and there should be no error when the result is non-nil. For every test where expected=false, we expect an error & nil return. It's probably overkill, but this is my pattern to check that the conditions on L164 are either both true or both false.